pkg/deploy/generator/resources.go

```diff
@@ -1089,6 +1078,116 @@ EOF
 
 chcon -R system_u:object_r:var_log_t:s0 /var/opt/microsoft/linuxmonagent
 
+cat >/usr/lib/systemd/system/download-mdm-credentials.service <<'EOF'
+[Unit]
+Description=Periodic mdm credentials refresh
+
+[Service]
+Type=oneshot
+ExecStart=/usr/local/bin/download-mdm-credentials.sh
+EOF
+
+cat >/usr/lib/systemd/system/download-mdm-credentials.timer <<'EOF'
+[Unit]
+Description=Periodic mdm credentials refresh
+
+[Timer]
+OnBootSec=0min
+OnCalendar=0/12:00:00
+AccuracySec=5s
+
+[Install]
+WantedBy=timers.target
+EOF
+
+cat >/usr/local/bin/download-mdm-credentials.sh <<EOF
+#!/bin/bash
+set -eu
+
+NEW_CERT_FILE="/etc/temp-mdm.pem"
+CURRENT_CERT_FILE="/etc/mdm.pem"
+
+az login -i
+az account set -s "$SUBSCRIPTIONID"
+
+rm -f \$NEW_CERT_FILE
+
+for attempt in {1..5}; do
+  az keyvault certificate download --file \$NEW_CERT_FILE --id "https://${KEYVAULTPREFIX}.vault.azure.net/certificates/test-cert" && break
+  if [[ \$attempt -lt 5 ]]; then sleep 10; else exit 1; fi
+done
+az logout
+
+if [ -f \$NEW_CERT_FILE ]; then
+  chmod 0600 \$NEW_CERT_FILE
+  sed -i -ne '1,/END CERTIFICATE/ p' \$NEW_CERT_FILE
+  mv \$NEW_CERT_FILE \$CURRENT_CERT_FILE
+else
+  echo Failed to refresh certificate && exit 1
+fi
+EOF
+chmod u+x /usr/local/bin/download-mdm-credentials.sh
+
+cat >/usr/lib/systemd/system/download-mdsd-credentials.service <<EOF
+[Unit]
+Description=Periodic mdsd certificate refresh
+
+[Service]
+Type=oneshot
+ExecStart=/usr/local/bin/download-mdsd-credentials.sh
+EOF
+
+cat >/usr/lib/systemd/system/download-mdsd-credentials.timer <<EOF
+[Unit]
+Description=Periodic mdsd certificate refresh
+
+[Timer]
+OnBootSec=0min
+OnCalendar=0/12:00:00
+AccuracySec=5s
+
+[Install]
+WantedBy=timers.target
+EOF
+
+cat >/usr/local/bin/download-mdsd-credentials.sh <<EOF
+#!/bin/bash
+set -eu
+
+NEW_CERT_FILE="/etc/temp-mdsd.pem"
+CURRENT_CERT_FILE="/etc/mdsd.pem"
+
+az login -i
+az account set -s "$SUBSCRIPTIONID"
+
+rm -f \$NEW_CERT_FILE
```

Comment:
`\$` is necessary because I need to expand `KEYVAULTPREFIX` and `SUBSCRIPTIONID` from original context, herefore we can't use `'EOF'` and have to escape the $